### ISSUE TITLE
no prefill when decoder_input_details=True from InferenceClient

### ISSUE THREAD

**Issue by lifeng-jin** [huggingface/text-generation-inference#2973]
### System Info

I used 3.0.2 official docker to load a local Llama 3 instruct model

### Information

- [x] Docker
- [ ] The CLI directly

### Tasks

- [x] An officially supported command
- [ ] My own modifications

### Reproduction

I used 3.0.2 official docker to load a local Llama 3 instruct model, and used InferenceClient to call it (see some interaction [here](https://github.com/huggingface/huggingface_hub/issues/2804))

```
output = client.text_generation("Today is a ", max_new_tokens=2, do_sample=True, temperature=1.0, details=True, decoder_input_details=True)
```

The output is this. Prefill is empty.
```
TextGenerationOutput(generated_text='5-minute', details=TextGenerationOutputDetails(finish_reason='length', generated_tokens=2, prefill=[], tokens=[TextGenerationOutputToken(id=20, logprob=-2.1425781, special=False, text='5'), TextGenerationOutputToken(id=24401, logprob=-4.4609375, special=False, text='-minute')], best_of_sequences=None, seed=9305067545921572115, top_tokens=None))

### Expected behavior

I expect prefill to include tokens in the prompt as well as their logprobs, as shown in the doc [here](https://huggingface.co/docs/huggingface_hub/en/package_reference/inference_client#huggingface_hub.InferenceClient.text_generation).

**Comment by sioan** [comment]
bump

**Comment by alexshtf** [comment]
+1

**Comment by alexshtf** [comment]
Turns out it has been disabled as of version 3.0:
https://huggingface.co/docs/text-generation-inference/en/reference/launcher#enableprefilllogprobs

Anybody from huggingface can please make sure this is properly documented in release notes as a **breaking change**? It breaks the work of many researchers who need logits of pre-filled responses. Happens a lot in RAG research, for example, where we want to use logits for various tasks regarding the retrieved passages.

**Comment by alexshtf** [comment]
@ydshieh  / @danieldk  - maybe you can help document it properly in the next release, or know who can do it?
